lib/decoder/real-time.js

```diff
@@ -0,0 +1,66 @@
+const url = require('url')
+
+// https://github.com/PRX/Infrastructure/blob/master/cdn/dovetail3-cdn.yml#L289
+const FIELDS = [
+  'timestamp',
+  'c-ip',
+  'sc-status',
+  'cs-method',
+  'cs-uri-stem',
+  'cs-user-agent',
+  'cs-referer',
+  'x-forwarded-for',
+  'sc-content-len',
+  'sc-range-start',
+  'sc-range-end',
+]
+const ASCII = /^[\x00-\xFF]*$/
+
+/**
+ * check for cloudfront real-time log events, with known TSV fields
+ */
+exports.detect = function (buffer) {
+  const str = buffer.toString('utf-8')
+  return ASCII.test(str) && str.split('\t').length === FIELDS.length
+}
+
+/**
+ * parse tab separated log lines
+ */
+exports.decode = async function (buffer) {
+  const parts = buffer.toString('utf-8').trim().split('\t')
```

Comment:
Good question!  For the "standard log" format, they [pretty explicitly](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/AccessLogs.html#LogFileFormat) say those characters will be url encoded.  But I don't see anything in the live-logs docs.

Also seems like clients should/would be encoding those, if you threw one in a header or something?  But will test, because that's good to know!

Reply:
Okay, confirmed both CW log formats url-encode tabs.  And other characters.  But I don't think there's any TODO on urlDecoding here yet, since the fields we're looking at won't have tabs (urls, query-params, etc).

If we someday start using the `User-Agent` / `Referer` / etc of the CDN requests ... would have to think about if/how to url-decode those.